lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -1957,6 +1957,99 @@ def plot(self, xs, ys, *args, zdir='z', **kwargs):
+        single value to be used for all points.
+
+        Parameters
+        ----------
+        x1, y1, z1 : float or 1D array-like
+            x, y, and z  coordinates of vertices for 1st line.
+
+        x2, y2, z2 : float or 1D array-like
+            x, y, and z coordinates of vertices for 2nd line.
+
+        where : array of bool (length N), optional
+            Define *where* to exclude some regions from being filled. The
+            filled regions are defined by the coordinates ``pts[where]``,
+            for all x, y, and z pts. More precisely, fill between ``pts[i]``
+            and ``pts[i+1]`` if ``where[i] and where[i+1]``. Note that this
+            definition implies that an isolated *True* value between two
+            *False* values in *where* will not result in filling. Both sides of
+            the *True* position remain unfilled due to the adjacent *False*
+            values.
+
+        single_polygon : bool
+            If True, then the 1st and 2nd lines are connected to form a single
+            polygon, which may be faster and render more cleanly for simple
+            shapes (eg, for filling between two lines that lie on a single
+            plane). If False (default), then a separate polygon is created for
+            each pair of subsequent points in the 1st and 2nd lines. This is
+            more flexible, but may be slower and result in rendering artifacts.
+
+        **kwargs
+            All other keyword arguments are passed on to `.Poly3DCollection`.
+
+        Returns
+        -------
+        `.Poly3DCollection`
+            A `.Poly3DCollection` containing the plotted polygons.
+
+        """
+        had_data = self.has_data()
+        x1, y1, z1, x2, y2, z2 = cbook._broadcast_with_masks(x1, y1, z1, x2, y2, z2)
+
+        if where is None:
+            where = True
+        else:
+            where = np.asarray(where, dtype=bool)
+            if where.size != x1.size:
+                raise ValueError(f"where size ({where.size}) does not match "
+                                 f"size ({x1.size})")
+        where = where & ~np.isnan(x1)  # NaNs were broadcast in _broadcast_with_masks
+
+        if single_polygon:
+            poly = []
+            for i in range(len(x1)):
+                if where[i]:
+                    poly.append((x1[i], y1[i], z1[i]))
+            for i in range(len(x2) - 1, -1, -1):
+                if where[i]:
+                    poly.append((x2[i], y2[i], z2[i]))
+            polys = [poly]
+
+        else:
+            polys = []
+            for idx0, idx1 in cbook.contiguous_regions(where):
+                x1slice = x1[idx0:idx1]
+                y1slice = y1[idx0:idx1]
+                z1slice = z1[idx0:idx1]
+                x2slice = x2[idx0:idx1]
+                y2slice = y2[idx0:idx1]
+                z2slice = z2[idx0:idx1]
+
+                if not len(x1slice):
+                    continue
+
```

Comment:
I'd put the "single_polygon" case in here, because you still want to break it by the contiguous regions I think? This will simplify the implementation a bit.

```python
            if single_polygon:
                poly = np.stack([np.concatenate([x1slice, x2slice[::-1]]),
                            np.concatenate([y1slice, y2slice[::-1]]),
                            np.concatenate([z1slice, z2slice[::-1]])]).T
                polys.append(poly)
            else:
...
```

Reply:
Yeah, I think you're right, this matches the 2D `fill_between` behavior better.